crates/sozo/ops/src/statistics.rs

```diff
@@ -1,45 +1,83 @@
 use std::fs::{self, File};
-use std::io::{self, BufReader};
+use std::io::{self, BufReader, Seek, SeekFrom};
 use std::path::PathBuf;
 
 use anyhow::{Context, Result};
+use cairo_lang_starknet_classes::casm_contract_class::CasmContractClass;
+use cairo_lang_starknet_classes::contract_class::ContractClass;
 use camino::Utf8PathBuf;
+use scarb_ui::Ui;
 use starknet::core::types::contract::SierraClass;
 use starknet::core::types::FlattenedSierraClass;
 
 #[derive(Debug, PartialEq)]
 pub struct ContractStatistics {
     pub contract_name: String,
-    pub number_felts: u64,
-    pub file_size: u64,
+
+    pub sierra_bytecode_size: u64,
+    pub sierra_contract_class_size: u64,
+
+    pub casm_bytecode_size: u64,
+    pub casm_contract_class_size: u64,
 }
 
-fn read_sierra_json_program(file: &File) -> Result<FlattenedSierraClass> {
-    let contract_artifact: SierraClass = serde_json::from_reader(BufReader::new(file))?;
+fn read_sierra_json_program(file: &File) -> Result<(FlattenedSierraClass, CasmContractClass)> {
+    let mut buf_reader = BufReader::new(file);
+    let contract_artifact: SierraClass = serde_json::from_reader(&mut buf_reader)?;
+    buf_reader.seek(SeekFrom::Start(0)).unwrap();
+
+    let contract_class: ContractClass = serde_json::from_reader(&mut buf_reader)?;
+    buf_reader.seek(SeekFrom::Start(0)).unwrap();
+
+    let casm_contract_class: CasmContractClass =
+        CasmContractClass::from_contract_class(contract_class, false, usize::MAX)?;
+
     let contract_artifact: FlattenedSierraClass = contract_artifact.flatten()?;
 
-    Ok(contract_artifact)
+    Ok((contract_artifact, casm_contract_class))
 }
 
 fn get_sierra_byte_code_size(contract_artifact: FlattenedSierraClass) -> u64 {
     contract_artifact.sierra_program.len() as u64
 }
 
+fn get_casm_byte_code_size(contract_artifact: CasmContractClass) -> u64 {
+    contract_artifact.bytecode.len() as u64
+}
+
 fn get_file_size(file: &File) -> Result<u64, io::Error> {
     file.metadata().map(|metadata| metadata.len())
 }
 
 fn get_contract_statistics_for_file(
     contract_name: String,
     sierra_json_file: File,
-    contract_artifact: FlattenedSierraClass,
+    sierra_class: FlattenedSierraClass,
+    casm_class: CasmContractClass,
 ) -> Result<ContractStatistics> {
-    let file_size = get_file_size(&sierra_json_file).context("Error getting file size")?;
-    let number_felts = get_sierra_byte_code_size(contract_artifact);
-    Ok(ContractStatistics { file_size, contract_name, number_felts })
+    let sierra_contract_class_size =
+        get_file_size(&sierra_json_file).context("Error getting file size")?;
+    let sierra_bytecode_size = get_sierra_byte_code_size(sierra_class);
+
+    let casm_contract_class_size = serde_json::to_string(&casm_class)
```

Comment:
There isn't a way to factorize how the byte size of a file is computed? (for Sierra and casm)

Reply:
fixed